File: app/src/main/java/com/rv150/musictransfer/model/Song.java
package com.rv150.musictransfer.model;

import android.os.Parcel;
import android.os.Parcelable;

public class Song implements Parcelable {
    public static final Parcelable.Creator<Song> CREATOR = new Parcelable.Creator<Song>() {
        @Override
        public Song createFromParcel(Parcel in) {
            return new Song(in.readString(), in.readString(), in.readLong());
        }

        @Override
        public Song[] newArray(int size) {
            return new Song[size];
        }
    };
    public final String title;
    public final String path;
    public final long size;

    public Song(String title, String path, long size) {
        this.title = title;
        this.path = path;
        this.size = size;
    }

    @Override
    public void writeToParcel(Parcel out, int flags) {
        out.writeString(title);
        out.writeString(path);
        out.writeLong(size);
    }

    @Override
    public int describeContents() {
        return 0;
    }
}

File: app/src/main/java/com/rv150/musictransfer/model/Song.kt
package com.rv150.musictransfer.model

import android.os.Parcel
import android.os.Parcelable

data class Song(val title: String, val path: String, val size: Long) : Parcelable {

    companion object {
        @JvmField @Suppress("unused")
        val CREATOR = object : Parcelable.Creator<Song> {
            override fun createFromParcel(source: Parcel?): Song {
                val parcel = source!!
                return Song(parcel.readString(), parcel.readString(), parcel.readLong())
            }

            override fun newArray(size: Int): Array<Song?> {
                return arrayOfNulls(size)
            }
        }
    }

    override fun writeToParcel(dest: Parcel?, flags: Int) {
        val parcel = dest!!
        parcel.writeString(title)
        parcel.writeString(path)
        parcel.writeLong(size)
    }

    override fun describeContents() = 0
}
